bionic/flake8/check_dnode_match.py

```diff
@@ -0,0 +1,220 @@
+"""
+This module contains a Flake8 plugin validating our use of `DescriptorNode.fail_match`.
+
+The goal is to check that we only call `fail_match` after exhaustively checking all
+cases, as in the following:
+
+    if dnode.is_entity():
+        ...
+
+    elif dnode.is_tuple():
+        ...
+
+    ...
+
+    else:
+        dnode.fail_match()
+
+The plugin requires this exact structure, although it doesn't care about what order the
+cases are handled in.
+
+The plugin's entry point is the Checker class; Flake8 loads this class based on the
+`flake8:local-plugins` config in `setup.cfg`, and then instantiates it with a parsed
+syntax tree for each file in our project.
+
+"""
+
+import ast
+from enum import Enum
+
+import attr
+
+from ..descriptors.ast import Subtype
+from ..utils.misc import oneline
+
+
+EXPECTED_TESTED_METHOD_NAMES = {f"is_{subtype.value}" for subtype in Subtype}
+N_EXPECTED_METHODS = len(EXPECTED_TESTED_METHOD_NAMES)
+EXPECATION_MSG = ""
```

Comment:
Do you need this?

Reply:
Nope, looks like it should have been deleted.